### PR TITLE
test(ARCH-458): rewrite tests forms 

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

we want to support react 18

**What is the chosen solution to this problem?**

rewrite tests using react-testing library of `@talend/react-forms`

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Review by romainseb — APPROVED** [review]

File: packages/forms/src/UIForm/fields/Text/Text.component.test.js [hunk 1]
```diff
@@ -138,53 +140,62 @@ describe('Text field', () => {
...
+		render(<Text {...props} />);
 		const event = { target: { value: 'totoa' } };
 
 		// when
-		wrapper.find('input').simulate('change', event);
+		screen.getByLabelText('My input title').focus();
+		await userEvent.keyboard('totoa');
+		screen.getByLabelText('My input title').blur();
+
+		// wrapper.find('input').simulate('change', event);
```
Comment (Nigui) [inline packages/forms/src/UIForm/fields/Text/Text.component.test.js:161]:
```suggestion
```

File: packages/forms/src/UIForm/fieldsets/CollapsibleFieldset/CollapsibleFieldset.component.test.js [hunk 2]
```diff
@@ -62,116 +66,113 @@ const defaultTitleMockData = {
...
-			/>,
+		render(
+			<WidgetContext.Provider value={widgets}>
+				<CollapsibleFieldset {...props} actions={actions} />
+			</WidgetContext.Provider>,
 		);
-
-		expect(wrapper.find('.panel-title button#action1').length).toBe(1);
-		expect(wrapper.find('.panel-title button#action2').length).toBe(1);
+		screen.debug();
```
Comment (Nigui) [inline packages/forms/src/UIForm/fieldsets/CollapsibleFieldset/CollapsibleFieldset.component.test.js:148]:
```suggestion
```

File: packages/forms/src/widgets/fields/Select/Select.test.js [hunk 3]
```diff
@@ -1,68 +1,42 @@
...
 		// when
-		const wrapper = mount(<Select {...props} />);
+		const { container } = render(<Select {...props} name="test-name" />);
 		// then
-		const select = wrapper.find('select').getDOMNode();
-		expect(select.getAttribute('class')).toEqual('form-control myCustomClass');
-		expect(select.getAttribute('name')).toEqual('test-name');
+		const select = screen.getByRole('combobox');
+		expect(select).toHaveClass('form-control myCustomClass', { exact: true });
+		expect(select).toHaveAttribute('name', 'test-name');
```
Comment (Nigui) [inline packages/forms/src/widgets/fields/Select/Select.test.js:22]: still required as we test snapshot right after?

**Review by Nigui — APPROVED** [review]
👏 
Only some code suggestion for forgotten lines

**Pull request merged**